parentheses/parentheses.go

```diff
@@ -0,0 +1,67 @@
+package parentheses
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strconv"
+)
+
+func IsBalanced(s string) bool {
+	stack := []rune{}
+	mapping := map[rune]rune{'(': ')', '[': ']', '{': '}'}
+
+	for _, c := range s {
+		if c == '(' || c == '[' || c == '{' {
+			stack = append(stack, c)
+		} else if c == ')' || c == ']' || c == '}' {
+			if len(stack) == 0 {
+				return false
+			}
+
+			last := stack[len(stack)-1]
+			stack = stack[:len(stack)-1]
+
+			if mapping[last] != c {
+				return false
+			}
+		}
+	}
+
+	return len(stack) == 0
+}
+
+func GenerateSequence(length int) string {
+	var parentheses string
```

Comment:
each time you're doing
```go
parentheses += ")"
```
you're creating a new string, and that also means, that GC will need to clean up the old string that's left behind after this operation. It's an inefficient operation if it's done over and over again.

use `strings.Builder` here